backend/models/node.py

```diff
@@ -1,87 +1,249 @@
-from .shared import db
-from .measurement import Measurement
+from .shared import db # Allows the models to be split out into separate files.
 
 
 class Node(db.Model):
-    __tablename__ = "node"
+    """
+    A class used to generate a table of Nodes.
+    Nodes are used to model Hierarchy trees that consist of objectives,
+    sub-objectives, and terminating in measurements.
+
+    Based off of SQLALchemy's adjacency_list.py:
+    https://docs.sqlalchemy.org/en/14/_modules/examples/adjacency_list/adjacency_list.html
+
+    ...
+
+    Attributes
+    ----------
+    id : int
+        Node's unique id.
+    parent_id : int
+        Parent node's unique id.
+    hierarchy_id : int
+        Id of the hierarchy the node belongs to.
+    name : str
+        The name of the node.
+    icon : str
+        String to store the icon the frontend uses for the node.
+    weight : float
+        Floating point value representing weight
+    is_measurement : bool
+        Flag for frontend.
+    measurement_type : str
```

Comment:
It still feels like measurement should exist as a class. Before I modeled nodes and measurements as a 1-to-many relationship, but that was wrong. Now, I think its actually a 1-to-1 relationship. Measurement should include the measure, local_value, global_value, measurement_type, and value_function. None of those fields are necessary to a normal node. Measurement would be null for most nodes, but a measurement node would have a measurement.